.travis.yml

```diff
@@ -7,18 +7,30 @@ git:
   quiet: true
 
 stages:
+  - name: preflight
+    if: type = pull_request
   - name: tests
-    if: type IN (push, pull_request)
+    if: type = pull_request
   - name: images
     if: type = pull_request AND branch = master
 
 jobs:
   include:
-    - stage: tests
+    - stage: preflight
       name: "Unit Tests"
       python: 3.6
       before_script: pip install tox-travis
       script: tox
+    - stage: preflight
+      services: docker
+      name: "Images build"
```

Comment:
This stage can be made optional by leveraging travis `PullRequestID` or `TRAVIS_BRANCH` variables.

Reply:
Sounds good.
I will look into it.

Reply:
We can leave this to later, I have the parts that are required to get tags from the PR description into the travis environment. The branch part could be solved via regex matching.